scripts/west_commands/runners/spi_burn.py

```diff
@@ -0,0 +1,70 @@
+import os
+import time
+import sys
+import subprocess
+
+from runners.core import ZephyrBinaryRunner, RunnerCaps
+
+TELINK_TOOLCHAIN_BASE = os.environ["TELINK_TOOLCHAIN_BASE"]
+SPI_BURN = f"{TELINK_TOOLCHAIN_BASE}/flash/bin/SPI_burn"
+ICEMAN_DIR = f"{TELINK_TOOLCHAIN_BASE}/ice"
+ICEMAN = "./ICEman"
+
+class SpiBurnBinaryRunner(ZephyrBinaryRunner):
+    '''Runner front-end for SPI_burn.'''
+
+    def __init__(self, cfg, addr, eraseall):
+        super().__init__(cfg)
+        self.bin = cfg.bin_file
+        self.addr = addr
+        if eraseall:
+            self.eraseall = "--erase-all"
+        else:
+            self.eraseall = ''
+
+    @classmethod
+    def name(cls):
+        return 'SPI_burn'
+
+    @classmethod
+    def capabilities(cls):
+        return RunnerCaps(commands={'flash'})
+
+    @classmethod
+    def do_add_parser(cls, parser):
+        parser.add_argument('--addr', default='0x0', 
+                            help='start flash address to write')
+        parser.add_argument('--erase-all', default=True,
+                            help='frace flash before burn')
+
+    @classmethod
+    def do_create(cls, cfg, args):
+        return SpiBurnBinaryRunner(cfg, args.addr, args.erase_all)
+
+    def do_run(self, command, **kwargs):
+        
+        origin_dir = os.getcwd()
+        try:
+
+            # Go into ICEman dir
+            os.chdir(ICEMAN_DIR)
+
+            # RUN ice in a background
+            cmd_ice_run = [ICEMAN, '-Z', 'v5', '-I', 'aice_sdp.cfg']
+
+            ice_process = subprocess.Popen(cmd_ice_run)
```

Comment:
What happens if the process is interrupted? Do you need to use popen_ignore_int, or is it OK if the user presses control-C and terminates this subprocess?

Reply:
I have tried control-C and it works as expected.